contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -149,11 +203,208 @@ contract MetaMorphoStrategy is ERC4626Strategy {
 
   /// @dev return always a value which is multiplied by 1e18
   /// eg for 2% apr -> 2*1e18
-  /// @notice this lending market is returning the apr already compounded (apy)
   function getApr() external view override returns (uint256 apr) {
+    apr = getAprWithLiquidityChange(0, 0);
+  }
+
+  /// @dev return always a value which is multiplied by 1e18
+  /// eg for 2% apr -> 2*1e18. Add and sub params are used off-chain
+  /// to calculate the variation of the apr based on the liquidity added/removed
+  function getAprWithLiquidityChange(uint256 add, uint256 sub) public view returns (uint256 apr) {
+    // TODO update mm snippet to include add and sub
+    // uint256 ratePerSecond = IMetamorphoSnippets(mmSnippets).supplyAPYVault(strategyToken, add, sub);
+
     uint256 ratePerSecond = IMetamorphoSnippets(mmSnippets).supplyAPYVault(strategyToken);
     // ratePerSecond is the rate per second scaled by 18 decimals
-    // (eg 32943060 -> 32943060 * 24 * 3600 * 365 * 100 / 1e18 = 0.103% apy)
-    apr = ratePerSecond * 365 days * 100;
+    // (eg 32943060 -> 32943060 * 24 * 3600 * 365 * 100 / 1e18 = 0.103% apr)
+    apr = ratePerSecond * 365 days * 100 + getRewardsApr(add, sub);
```

Comment:
`ratePerSecond` is an APY, meaning it compounds interest annually (so it's not a rate per second, but a rate per year already): https://github.com/morpho-org/morpho-blue-snippets/blob/45c5d2e41e3577023e1c1b27b59957dc4d5bd316/src/metamorpho/MetaMorphoSnippets.sol#L112

If you're based on the previous version of the snippets, there was an error in it: https://github.com/morpho-org/morpho-blue-snippets/blob/ca1a1c88b939c482a08c3e05d8ea777ea009a689/src/metamorpho/MetaMorphoSnippets.sol#L116-L124

It is compounded twice and represents a quantity that is not easy to compare to others, so I'd recommend basing your work on the latest version of snippets. Additionally I would recommend including it in this contract.

Reply:
We are changing the way `supplyAPY` vault is behaving in a fork of the snippets (not yet ready) but basically we are removing `.wTaylorCompounded(365 days)` and this will be transformed into a `supplyAPRVault` to be consistent with all other strategies that we have. I considered having the 2 methods of the snippets inside this contract directly but it was a bit complicated to include everything, but will reevaluate this